### ISSUE TITLE
ClassificationModel giving same output for Binary Classification

### ISSUE THREAD

**Issue by iknoorjobs** [ThilinaRajapakse/simpletransformers#389]
Hi!

I am trying to run a simple binary classification with default parameters. But when I do the evaluation, it gives the same output for all inputs. It's either all zeroes or ones. And when I evaluate metrics, it turns out the accuracy is the number of zeroes or ones present in the test set (eval_df). Why is this happening? 

I tried reducing epoch, but still no use. Also, my dataset is balanced. Still, can't figure out what's the problem. 

Please help. Thanks.

**Comment by ThilinaRajapakse** [comment]
I know you are running with the defaults, but can you still show the script just to check?

**Comment by iknoorjobs** [comment]
Hi!

Sure! 

`from simpletransformers.classification import ClassificationModel
import apex
train_args = { 'num_train_epochs': 2,  'overwrite_output_dir': True}
model = ClassificationModel('roberta', 'roberta-large', use_cuda=True, args=train_args)
model.train_model(train_df)
result, model_outputs, wrong_predictions = model.eval_model(eval_df)
print(result)
print(model_outputs)
`

model_outputs
[-0.6147461  -0.01327515]
 [-0.6142578  -0.01331329]
 [-0.6142578  -0.01332092]
 ...
 [-0.6147461  -0.01325226]
 [-0.6147461  -0.01324463]
 [-0.6147461  -0.01325226]

**Comment by ThilinaRajapakse** [comment]
Try setting `"reprocess_input_data": True` in the `train_args`.

**Comment by iknoorjobs** [comment]
Ok, Thanks. I am trying but the training is much slower now. Why does setting reprocess_input_data to True increases training time?

**Comment by ThilinaRajapakse** [comment]
The time taken for an epoch shouldn't change unless you are doing evaluation during training. However, this will add some overhead before the training starts as the features are computed as opposed to using cached features.

I suspected the issue is being caused by some old features being read from cache which is why I suggested doing `reprocess_input_data`.

**Comment by iknoorjobs** [comment]
Hi @ThilinaRajapakse 

Thanks! I have tried but still the same problem.

**Comment by iknoorjobs** [comment]
I ran the example on docs. The problem persists. It looks like there is some problem in the code. Check this colab,
Link: https://colab.research.google.com/drive/1DJcWCKBNge0jAOQ798eXm1aF14gTKGWD?usp=sharing

**Comment by iknoorjobs** [comment]
@ThilinaRajapakse Did you find any bug? I have been continuously trying but it's still giving same output for the test data.

**Comment by ThilinaRajapakse** [comment]
The output changes if you change the sentence in the last predict call of the notebook you shared.

**Comment by iknoorjobs** [comment]
@ThilinaRajapakse I guess, the default learning rate was not working with my dataset. I tried to reduce it to 1e-5 or less, and now its showing different outputs and much better results.

**Comment by ThilinaRajapakse** [comment]
That can happen. If the learning rate is too high or if you train the model for too long, Transformer models seem to completely break down. :man_shrugging:

**Comment by iknoorjobs** [comment]
Exactly, Thanks a lot. :)
